mxcube3/ui/components/PopInput/DefaultInput.jsx

```diff
@@ -8,14 +8,19 @@ export default class DefaultInput extends React.Component {
 
   constructor(props) {
     super(props);
+    this.state = { value: this.props.value };
     this.save = this.save.bind(this);
     this.cancel = this.cancel.bind(this);
     this.submit = this.submit.bind(this);
+    this.onChange = this.onChange.bind(this);
   }
 
+  onChange(event) {
+    this.setState({ value: event.target.value });
```

Comment:
I've already made a fix for this as well I have not made the PR yet though. My fix does not involve setState, I think we should avoid setState when we can. Is there any reason for using it here except for the obvious one to retrieve the value ?